nerfstudio/utils/math.py

```diff
@@ -285,3 +286,26 @@ def safe_normalize(
         Normalized vectors.
     """
     return vectors / (torch.norm(vectors, dim=-1, keepdim=True) + eps)
+
+
+def masked_reduction(loss: TensorType[0], M: TensorType[0], reduction_type: Literal["image", "batch"]):
+    """
+    Whether to consolidate the loss across the batch or across the image
+    Args:
+        loss: loss tensor
+        M: mask tensor
```

Comment:
Rename `loss` -> `input_tensor`, and `M` -> `mask` and update docs to not be loss specific.